### PR TITLE
ENH: Bump ITK workflows for v5.3rc04.post4.

### LINKED ISSUE #139: Update to Windows-2022?
Should we update this repository to use `Windows-2022` in CI for compatibility with latest wheels?

### PR DESCRIPTION
**Pull request by tbirdso**
Updates cookiecutter CI workflow to rely on
ITKRemoteModuleBuildTestPackageAction reusable workflows for easier maintenance.

Updates ITK reference to v5.3rc04.post4 and applies version updates from reusable workflow.

See https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/pull/9. Closes #139.

### CODE REVIEW

File: .github/workflows/build-test-package.yml [hunk 1]
```diff
@@ -61,12 +61,12 @@ jobs:
...
 
     - name: Build ITK
-      if: matrix.os == 'windows-2019'
+      if: matrix.os == 'windows-2022'
       run: |
         cd ..
         mkdir ITK-build
         cd ITK-build
-        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+        call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:69]: I tried going to 2022 in https://github.com/dzenanz/ITKIOOMEZarrNGFF/tree/tensorstore, but after realizing it is not working I reverted back to 2019. Now I realize that rename `2019`->`2022` is not the only thing here, but change from `Program Files (x86)` to `Program Files` (64-bit).

File: .github/workflows/build-test-package.yml [hunk 2]
```diff
@@ -156,6 +156,12 @@ jobs:
         sudo rm -rf "$AGENT_TOOLSDIRECTORY"
         df -h
 
+    - name: 'Fetch build dependencies'
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:159]: By looking at log of the run, I realized that this was not needed - `ubuntu-22.04` already includes zstd and it is the latest version.

File: .github/workflows/build-test-package.yml [hunk 3]
```diff
@@ -3,22 +3,22 @@ name: Build, test, package
...
+  itk-wheel-tag: "v5.3rc04.post4"
 
 jobs:
   build-test-cxx:
     runs-on: ${{ matrix.os }}
     strategy:
       max-parallel: 3
       matrix:
-        os: [ubuntu-20.04, windows-2019, macos-11]
+        os: [ubuntu-20.04, windows-2022, macos-11]
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:15]: Should we bump this to `ubuntu-22.04`?

File: {{cookiecutter.project_name}}/.github/workflows/build-test-package.yml [hunk 4]
```diff
@@ -2,281 +2,11 @@ name: Build, test, package
...
-      run: |
-        ls -R
-        for d in */; do
-          mv ${d}/*.whl .
-        done
-        mkdir dist
-        mv *.whl dist/
-        ls dist
+  cxx-build-workflow:
+    uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-cxx.yml@v5.3rc04.post4
```
Comment (dzenanz) [inline {{cookiecutter.project_name}}/.github/workflows/build-test-package.yml:7]: Great simplification here, it should make our life a lot easier in the future.

File: .github/workflows/build-test-package.yml [hunk 5]
```diff
@@ -238,11 +244,11 @@ jobs:
         path: Evaluated/ITKModuleTemplate/dist
 
   build-windows-python-packages:
-    runs-on: windows-2019
+    runs-on: windows-2022
     strategy:
       max-parallel: 2
       matrix:
-        python-version-minor: ["7", "8", "9", "10"]
+        python-version-minor: ["7", "8", "9", "10", "11"]
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:251]: It is kind of good that this module itself uses the verbose version of the CI specification, so the more complicated remotes have a template to follow during updating.

**Review by dzenanz — APPROVED** [review]

File: .github/workflows/build-test-package.yml (hunk 2 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:159]: I'm not sure I understand, the most recent run uses `ubuntu-20.04` and not `ubuntu-22.04`?

File: .github/workflows/build-test-package.yml (hunk 5 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:251]: CI for this module also requires the additional step of evaluating the cookiecutter template, so we won't be able to easily replace with a reusable workflow.

File: .github/workflows/build-test-package.yml (hunk 1 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:69]: Note that the bump for v5.3rc04.post4 contains a couple of other small changes as well, see the commit history at https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/pull/9/commits

File: .github/workflows/build-test-package.yml (hunk 2 above)
Comment (dzenanz) [reply .github/workflows/build-test-package.yml:159]: I used 22.04 in ome-ngff module. In 20.04 apt says: `The following NEW packages will be installed: zstd`.

File: .github/workflows/build-test-package.yml (hunk 3 above)
Comment (dzenanz) [reply .github/workflows/build-test-package.yml:15]: OK, if CI comes green, merge.

File: .github/workflows/build-test-package.yml (hunk 2 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:159]: Okay great, good catch 👍 let's stick with `ubuntu-20.04` for stability for now and plan to remove the zstd install step when we bump to `ubuntu-22.04` in the near future.

File: .github/workflows/build-test-package.yml (hunk 5 above)
Comment (dzenanz) [reply .github/workflows/build-test-package.yml:251]: Some other remote modules require extra stuff (e.g. Ultrasound requires many things, ome-ngff requires a few unusual build tools such as `perl` and `nasm`).

File: .github/workflows/build-test-package.yml (hunk 3 above)
Comment (tbirdso) [reply .github/workflows/build-test-package.yml:15]: Added https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/issues/10 to track ubuntu upgrade

**Review by thewtex — APPROVED** [review]
🚀

**Pull request merged**